adapter/utils/src/test/java/org/mobilitydata/gtfsvalidator/timeutils/TimeConversionUtilsTest.java

```diff
@@ -0,0 +1,198 @@
+ *  limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.timeutils;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.Assert.*;
+
+public class TimeConversionUtilsTest {
+    private static final int HOUR_TO_SEC_CONVERSION_FACTOR = 3600;
+    private static final int NOON = (12 * HOUR_TO_SEC_CONVERSION_FACTOR);
+    private static final int MIN_TO_SEC_CONVERSION_FACTOR = 60;
+    private static final int SEC_TO_SEC_CONVERSION_FACTOR = 1;
+
+    // Remove warning "PointlessArithmeticExpression" since they these expressions are written to ease code
+    // comprehension
+    @SuppressWarnings("PointlessArithmeticExpression")
+    @Test
+    void hourInHHMMSSShouldConvertToIntegerFromNoon() {
+        final String threePm = "15:00:00";
+        int toCheck = new TimeConversionUtils().convertHHMMSSToIntFromNoonOfDayOfService(threePm);
+        assertEquals((15 * HOUR_TO_SEC_CONVERSION_FACTOR +
+                0 * MIN_TO_SEC_CONVERSION_FACTOR + 0 * SEC_TO_SEC_CONVERSION_FACTOR) - NOON, toCheck);
+
+        final String noon = "12:00:00";
+        toCheck = new TimeConversionUtils().convertHHMMSSToIntFromNoonOfDayOfService(noon);
+        assertEquals(((NOON + 0 * MIN_TO_SEC_CONVERSION_FACTOR + 0 * SEC_TO_SEC_CONVERSION_FACTOR) - NOON), toCheck);
+
+        final String midnight = "24:00:00";
+        toCheck = new TimeConversionUtils().convertHHMMSSToIntFromNoonOfDayOfService(midnight);
+        assertEquals(((24 * HOUR_TO_SEC_CONVERSION_FACTOR + 0 * MIN_TO_SEC_CONVERSION_FACTOR +
+                0 * SEC_TO_SEC_CONVERSION_FACTOR) - NOON), toCheck);
+
+        final String sixFortyPm = "18:40:00";
+        toCheck = new TimeConversionUtils().convertHHMMSSToIntFromNoonOfDayOfService(sixFortyPm);
+        assertEquals(((18 * HOUR_TO_SEC_CONVERSION_FACTOR + 40 * MIN_TO_SEC_CONVERSION_FACTOR + 0) - NOON), toCheck);
+
+        final String tenAm = "10:00:00";
+        toCheck = new TimeConversionUtils().convertHHMMSSToIntFromNoonOfDayOfService(tenAm);
+        assertEquals(((10 * HOUR_TO_SEC_CONVERSION_FACTOR + 0 * MIN_TO_SEC_CONVERSION_FACTOR +
+                0 * SEC_TO_SEC_CONVERSION_FACTOR) - NOON), toCheck);
+
+        final String sixThirtyAm20sec = "06:30:20";
+        toCheck = new TimeConversionUtils().convertHHMMSSToIntFromNoonOfDayOfService(sixThirtyAm20sec);
+        assertEquals(((6 * HOUR_TO_SEC_CONVERSION_FACTOR + 30 * MIN_TO_SEC_CONVERSION_FACTOR +
+                20 * SEC_TO_SEC_CONVERSION_FACTOR) - NOON), toCheck);
+
+        final String oneThirtyAm40sec = "25:30:40";
+        toCheck = new TimeConversionUtils().convertHHMMSSToIntFromNoonOfDayOfService(oneThirtyAm40sec);
+        assertEquals(((25 * HOUR_TO_SEC_CONVERSION_FACTOR + 30 * MIN_TO_SEC_CONVERSION_FACTOR +
+                40 * SEC_TO_SEC_CONVERSION_FACTOR) - NOON), toCheck);
+    }
+
+    @Test
+    void callConvertHHMMSSToIntFromNoonOfDayOfServiceOnNullValueShouldReturnNull() {
```

Comment:
Would remove the `call` prefix as it doesn't provide any meaningful information

Reply:
Done in  789d7a6e991b90167babdc2bca5423a62f623817